enterprise/server/experiments/experiments_test.go

```diff
@@ -111,6 +114,71 @@ func writeFlagConfig(t testing.TB, data string) string {
 	return path
 }
 
+// TestStablePercentage ensures that once a parameter is included in an
+// experiment, it remains included even as the percentage increases.
+func TestStablePercentage(t *testing.T) {
+	var testFlags = `{
+  "$schema": "https://flagd.dev/schema/v0/flags.json",
+  "flags": {
+    "percent-experiment": {
+      "state": "ENABLED",
+      "defaultVariant": "false",
+      "variants": {
+        "true": true,
+        "false": false
+      },
+      "targeting": {
+        "fractional": [
+			{
+				"cat": [{ "var": "$flagd.flagKey" }, { "var": "experiment-param" }]
+			},
+			["true", %d],
+			["false", %d]
+		]
+      }
+    }
+  }
+}
+`
+	offlineFlagPath := writeFlagConfig(t, fmt.Sprintf(testFlags, 0, 100))
+	writePercentConfig := func(percent int) {
+		os.WriteFile(offlineFlagPath, []byte(fmt.Sprintf(testFlags, percent, 100-percent)), os.ModePerm)
+	}
+
+	ctx := context.Background()
+	params := make([]string, 100)
+	values := make([]bool, 100)
+	for i := range params {
+		p, err := random.RandomString(10)
+		require.NoError(t, err)
+		params[i] = p
+	}
+	for percent := 0; percent <= 100; percent++ {
+		writePercentConfig(percent)
+
+		// Create a new provider after each write. Otherwise the test can race
+		// with the provider's internal goroutine that reads the file.
+		provider := flagd.NewProvider(flagd.WithInProcessResolver(), flagd.WithOfflineFilePath(offlineFlagPath))
```

Comment:
would this work across apps? wondering what mechanism it uses to ensure stability. Is it hashing the experiment context or something like that? and if so, is there anything unstable in our experiment context currently which might thwart the hashing?